processor/k8sattributesprocessor/testdata/e2e/collector/deployment.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Name }}
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: opentelemetry-collector
+      app.kubernetes.io/instance: {{ .Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: opentelemetry-collector
+        app.kubernetes.io/instance: {{ .Name }}
+    spec:
+      serviceAccountName: {{ .Name }}
+      containers:
+        - name: opentelemetry-collector
+          command:
+            - /otelcontribcol
+            - --config=/conf/relay.yaml
+          image: "otelcontribcol:latest"
+          imagePullPolicy: Never
+          ports:
+            - name: otlp
+              containerPort: 4317
+              protocol: TCP
+          env:
+            - name: MY_POD_IP
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: status.podIP
+          livenessProbe:
+            httpGet:
+              path: /
+              port: 13133
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 13133
+          resources:
+            limits:
+              cpu: 128m
+              memory: 256Mi
```

Comment:
Do we need memory limit? I'm concerned that if the test keeps growing more complex with more test data to process, is there any possibility that collector maybe OOM?

If we do not specify the memory limit, I think collector could use memory as much as it can.

Reply:
I'd rather keep it as it's a good practice, and if the collector runs out of memory for this test case, it'll indicate that something is wrong